_test/test_sqw/test_PixelData.m

```diff
@@ -0,0 +1,149 @@
+classdef test_PixelData < TestCase
+
+properties
+    raw_pix_data = rand(9, 10);
+    pixel_data_obj;
+end
+
+methods (Access = private)
+
+    function pix_data = get_random_pix_data_(~, rows)
+        data = rand(9, rows);
+        pix_data = PixelData(data);
+    end
+
+end
+
+methods
+
+    function obj = test_PixelData(~)
+        obj = obj@TestCase('test_PixelData');
+
+        obj.pixel_data_obj = PixelData(obj.raw_pix_data);
+    end
+
+    function test_default_construction_sets_empty_pixel_data(~)
+        pix_data = PixelData();
+        num_cols = 9;
+        assertEqual(pix_data.data, zeros(num_cols, 0));
+    end
+
+    function test_PIXELDATA_raised_on_construction_with_data_with_lt_9_cols(~)
+        f = @() PixelData(ones(3, 3));
+        assertExceptionThrown(f, 'PIXELDATA:data_error')
+    end
+
+    function test_PIXELDATA_raised_on_construction_with_data_with_gt_9_cols(~)
+        f = @() PixelData(ones(10, 3));
+        assertExceptionThrown(f, 'PIXELDATA:data_error')
+    end
+
+    function test_coordinates_returns_empty_array_if_pixel_data_empty(~)
+        pix_data = PixelData();
+        assertTrue(isempty(pix_data.coordinates));
+    end
+
+    function test_pixel_data_is_set_to_input_data_on_construction(obj)
+        assertEqual(obj.pixel_data_obj.data, obj.raw_pix_data);
+    end
+
+    function test_coordinate_data_can_be_retrieved_using_get_function(obj)
+        assertEqual(get(obj.pixel_data_obj, 'coordinates'), ...
+                    obj.pixel_data_obj.coordinates);
+    end
+
+    function test_coordinate_data_can_be_set_using_set_function(obj)
+        num_rows = 10;
+        pix_data_obj = obj.get_random_pix_data_(num_rows);
+        new_coord_data = ones(4, num_rows);
+        set(pix_data_obj, 'coordinates', new_coord_data);
```

Comment:
is this equivalent to `pixel_data_obj.set('coordinates', new_coord_data)` ?

Reply:
Yes